Fix BUILD_CAFFE2 if FBGEMM and NNPACK are not built

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45610 Fix BUILD_CAFFE2 if FBGEMM and NNPACK are not built**

Also add to the usual documentation places that this option exists.

Differential Revision: [D24058199](https://our.internmc.facebook.com/intern/diff/D24058199)

Commit:
Fix BUILD_CAFFE2 if FBGEMM and NNPACK are not built

Also add to the usual documentation places that this option exists.

[ghstack-poisoned]

walterddr commented on caffe2/utils/CMakeLists.txt:
```diff
@@ -1,9 +1,13 @@
     utils/threadpool/ThreadPool.cc
   )
+
+  if(USE_PTHREADPOOL)
+    list(APPEND Caffe2_CPU_SRCS utils/threadpool/pthreadpool-cpp.cc)
+  endif()
```
we should probably do:
```suggestion
  if(USE_PTHREADPOOL AND NOT USE_INTERNAL_PTHREADPOOL_IMPL)
    list(APPEND Caffe2_CPU_SRCS utils/threadpool/pthreadpool-cpp.cc)
  endif()
```

walterddr commented on CONTRIBUTING.md:
```diff
@@ -489,8 +489,7 @@ only interested in a specific component.
   `make` if you don't have ninja installed).
-- Don't need Caffe2?  Pass `BUILD_CAFFE2_OPS=0` to disable build of
-  Caffe2 operators.
+- Don't need Caffe2?  Pass `BUILD_CAFFE2=0` to disable Caffe2 build.
```
BUILD_CAFFE2=0 also disable other stuff. should we keep both in the doc? (IMO we can put doc optimization in a separate PR?)

suo replied:
So BUILD_CAFFE2 is not a superset of BUILD_CAFFE2_OPs? If so, it should be noted in this contributing doc.

walterddr replied:
it is actually. what I meant is to keep both lines

walterddr replied:
e.g. 
```suggestion
- Don't need Caffe2 at all?  Pass `BUILD_CAFFE2=0` to disable Caffe2 build.
- Don't need Caffe2 ops?  Pass `BUILD_CAFFE2_OPS=0` to disable Caffe2 operator build.
```

walterddr replied:
although I am a bit hesitate to put this on the CONTRIBUTING.md - BUILD_CAFFE2=OFF doesn't necessarily disable ALL caffe2 builds at the moment, there are still some separation work to be done. 

I wouldn't block merging the PR though.

Commit:
Update on "Fix BUILD_CAFFE2 if FBGEMM and NNPACK are not built"


Also add to the usual documentation places that this option exists.

[ghstack-poisoned]
